From a2898eaaab2d08d98c9ba153a00b0beeca81643c Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Tue, 21 May 2024 14:20:05 +0800
Subject: [PATCH 25/67] PHYTIUM: net/phytmac: Bugfix MAC 10/100M speed does't
 work problem

Gigabit_mode_enable bit should be clear when the speed is switched
to 10M/100M.

Mainline: NA
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I16e54d6e173c15c890788cf0bdcff2facca0f490

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/6b70dbeaafdc06a66410e77bfba11c5595f25a4e
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac.h         | 2 +-
 drivers/net/ethernet/phytium/phytmac_ethtool.c | 2 +-
 drivers/net/ethernet/phytium/phytmac_v1.c      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac.h b/drivers/net/ethernet/phytium/phytmac.h
index cb9359c7bd4d..bc0d2670c2f5 100644
--- a/drivers/net/ethernet/phytium/phytmac.h
+++ b/drivers/net/ethernet/phytium/phytmac.h
@@ -13,7 +13,7 @@

 #define PHYTMAC_DRV_NAME		"phytium-mac"
 #define PHYTMAC_DRV_DESC		"PHYTIUM Ethernet Driver"
-#define PHYTMAC_DRIVER_VERSION		"1.0.2"
+#define PHYTMAC_DRIVER_VERSION		"1.0.3"
 #define PHYTMAC_DEFAULT_MSG_ENABLE	  \
 		(NETIF_MSG_DRV		| \
 		NETIF_MSG_PROBE	| \
diff --git a/drivers/net/ethernet/phytium/phytmac_ethtool.c b/drivers/net/ethernet/phytium/phytmac_ethtool.c
index 4f632591142f..a97a4c28c04f 100644
--- a/drivers/net/ethernet/phytium/phytmac_ethtool.c
+++ b/drivers/net/ethernet/phytium/phytmac_ethtool.c
@@ -429,7 +429,7 @@ static int phytmac_set_link_ksettings(struct net_device *ndev,
 	int ret = 0;

 	if (!ndev->phydev) {
-		netdev_err(ndev, "fixed link interface not supported set link\n");
+		netdev_err(ndev, "Without a PHY, setting link is not supported\n");
 		ret = -EOPNOTSUPP;
 	} else {
 		phy_ethtool_set_link_ksettings(ndev, kset);
diff --git a/drivers/net/ethernet/phytium/phytmac_v1.c b/drivers/net/ethernet/phytium/phytmac_v1.c
index 1d7c2e175917..20595aca1783 100644
--- a/drivers/net/ethernet/phytium/phytmac_v1.c
+++ b/drivers/net/ethernet/phytium/phytmac_v1.c
@@ -182,7 +182,7 @@ static int phytmac_mac_linkup(struct phytmac *pdata, phy_interface_t interface,

 	config = PHYTMAC_READ(pdata, PHYTMAC_NCONFIG);

-	config &= ~(PHYTMAC_BIT(SPEED) | PHYTMAC_BIT(FD));
+	config &= ~(PHYTMAC_BIT(SPEED) | PHYTMAC_BIT(FD) | PHYTMAC_BIT(GM_EN));

 	if (speed == SPEED_100)
 		config |= PHYTMAC_BIT(SPEED);
--
2.50.1
